template/{{.input_root_dir}}/{{template `project_name_alphanumeric_underscore` .}}/deployment/batch_inference/predict.py.tmpl

```diff
@@ -11,21 +11,22 @@ def predict_batch(
     """
     {{ if (eq .input_include_models_in_unity_catalog "yes") }}mlflow.set_registry_uri("databricks-uc"){{ end }}
     table = spark_session.table(input_table_name)
-    {{ if (eq .input_include_feature_store `yes`) }}
-    from databricks.feature_store import FeatureStoreClient
-    
+    {{ if (eq .input_include_feature_store `yes`) }} 
+    {{ if (eq .input_include_models_in_unity_catalog "no") }} 
+    from databricks.feature_store import FeatureStoreClient    
     fs_client = FeatureStoreClient()
-
-    prediction_df = fs_client.score_batch(
-        model_uri,
-        table
-    )
+    prediction_df = fs_client.score_batch(model_uri, table)    
+    {{ else }}   
+    from databricks.feature_engineering import FeatureEngineeringClient    
+    fe_client = FeatureEngineeringClient()
+    prediction_df = fe_client.score_batch(model_uri = model_uri, df = table)
+    {{ end }} 
     output_df = (
         prediction_df.withColumn("prediction", prediction_df["prediction"])
         .withColumn("model_version", lit(model_version))
         .withColumn("inference_timestamp", to_timestamp(lit(ts)))
     )
-    {{ else }}
+    {{ else }} 
```

Comment:
nit space

Reply:
see latest updates -- will be adding links to screenshots